week02/ImplementStrStr.cpp

```diff
@@ -0,0 +1,53 @@
+// TC: O(n)
+// MC: O(n)
+class Solution {
+public:
+    long long pow[10005];
+    long long prefixHash[10005];
```

Comment:
You are always creating an array of 10000 elements, regardless of the input size. It is not a clever approach and a memory hungry one. Replace it with vectors and resize them when necessary.